### DISCUSSION TITLE
Freeplane AI tutor

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/freeplane#1524] [Announcements]
You can ask this [Freeplane AI Tutor](https://chat.openai.com/g/g-CkkQa1aEP-freeplane-tutor ) any questions about Freeplane. It is provided with knowledge from Freeplane documentation, menus, and preferences.

Open AI [GPT Plus](https://help.openai.com/en/articles/6950777-what-is-chatgpt-plus) subscription required.

Let me know your impressions.

**Comment by sifran-github** [comment]
It can be really nice. I'm also looking forward to see how useful it is. Maybe it can answer all the basic questions in this forum.

also:

![image](https://github.com/freeplane/freeplane/assets/5730355/a0d12bbd-5ac6-40c5-9c8c-f3b286ecc627)

Maybe we can create another agent that is specializing in freeplane scripting?
@macmarrum @dpolivaev 

Just a few days ago I was asking chatgpt to code some simple examples for freeplane, and it failed. Gave me some API calls that did not exist, even when I asked very simple stuff.

**Reply by euu2021** [reply]
> Just a few days ago I was asking chatgpt to code some simple examples for freeplane, and it failed. Gave me some API calls that did not exist, even when I asked very simple stuff.

Some time ago I tried to use ChatGPT to create FP scripts, and it initially makes a lot of mistakes, but then I spent some time teaching it, and it improved. I explained to ChatGPT like if I was explaining to a human, and it learned. 

It was just a small experiment, though. I don't know how deep it can go.

**Reply by dpolivaev** [reply]
I tried to create a custom GPT knowing Freeplane API but I failed. If you were successful at teaching ChatGPT to create FP scripts, please share how you did it.

**Reply by euu2021** [reply]
For example, I asked for a script that changes the text of the node to "important", and ChatGPT created it wrongly; then I corrected, and the next script it created correctly, right away:
![image](https://github.com/freeplane/freeplane/assets/77707706/4e61f067-cd26-4748-a24a-edc145ca63a4)

**Reply by dpolivaev** [reply]
After playing around with custom GPT builder I came to the conclusion that now writing good documentation completely covering  Freeplane API and having examples and explanations would be very helpful. Because it would allow custom GPT to answer the questions correctly. The current documentation available in the source files is not sufficient and gives GPT too much space for guesses and hallucinations.

And because not enough people have paid subscriptions, I would have to use the OpenAI API and cover the expenses (I do not expect them to be high unless someone starts misusing it, and in this case, I would have to shut down the service). I do not think it is urgent, but it is interesting to try.

**Reply by euu2021** [reply]
This will be nice!

I can go into [this project](https://github.com/freeplane/docs/discussions/18#discussion-4117965) of creating one simple example of each method of the API. Maybe, putting all the methods in a Google Sheet, so multiple people can work on it simultaneously.

Would it be relevant, or there are more efficient ways of attacking the problem?

**Reply by dpolivaev** [reply]
From now writing comprehensive documentation starts to make sense because AI can read it and extract information for humans. The format is not relevant, but the better it is written, the more complete it is and the more correct examples it has, the better suggestions can be generated by the AI.

I do not expect too many people to write it simultaneously, so synchronization is not an issue. I think systematic API coverage in https://github.com/freeplane/docs markdown files is a good way to go. For instance, Freeplane Tutor AI works pretty well and it is provided only with all concatenated markdown files from this project and the Freeplane documentation mind map saved as plain text.

**Reply by euu2021** [reply]
I created a thread at the documentation repository: https://github.com/freeplane/docs/discussions/135

**Comment by sifran-github** [comment]
Chatgpt: "The latest version of Freeplane for which I have documentation is 1.9.15​​."
but it does know some things about some newer features, like layouts..

I even asked him why is that:
![image](https://github.com/freeplane/freeplane/assets/5730355/6213b739-728c-4f7c-a397-0ca35915fa00)

**Comment by sifran-github** [comment]
OK, This AI officially helped me with something that I didn't know. I knew that you can see the created/modified using mouse hover. but...

![image](https://github.com/freeplane/freeplane/assets/5730355/c3c067b6-24b6-4678-84cb-aedcbb705592)
It works, and I didn't know about it.

and by the way, if i search "created" or "modified" in the command search, it doesn't find anything (because its not AI, its searching exact matches):
![image](https://github.com/freeplane/freeplane/assets/5730355/1db509db-78dd-47c6-8063-a0f6e24cdfee)

**Reply by sifran-github** [reply]
just for the record, he knows the other way as well:
![image](https://github.com/freeplane/freeplane/assets/5730355/9bf7b1a9-ed68-42e6-850d-64716825cfbd)

**Comment by AliHmaou** [comment]
Just discovered it and already gave me solutions through groovy code !
Great idea !

**Comment by AliHmaou** [comment]
Just discovered it and already gave me solutions through groovy code !
Great idea !
